ir/memory.cpp

```diff
@@ -1518,7 +1521,19 @@ pair<expr,Pointer> Memory::refined(const Memory &other,
   expr offset = ptr.get_offset();
   expr ret(true);
 
-  for (unsigned bid = 1; bid < num_nonlocals; ++bid) {
+  auto is_constglb = [](const Memory &m, unsigned bid) {
+    return
+      find(m.non_local_blk_nonwritable.begin(),
+           m.non_local_blk_nonwritable.end(), bid) !=
+        m.non_local_blk_nonwritable.end();
+  };
+
+  for (unsigned bid = 1; bid < IR::num_nonlocals; ++bid) {
+    if (!is_constglb(*this, bid) && is_constglb(other, bid)) {
```

Comment:
this check can be replaced with num_nonlocals_src or something like that?

Reply:
this case can only happen between num_nonlocals_src and num_nonlocals_tgt. It needs an assert the very least.

Reply:
Updated